### ISSUE TITLE
DNF Rule "Specify Version" Not Applying to dnf-plugins-core

### ISSUE THREAD

**Issue by oliverisaac** [hadolint/hadolint#1002]
- [x] This is a bug report
- [x] I searched existing issues before opening this one

### Expected behavior

I would expect a docker line of:

```
RUN dnf install -y dnf-plugins-core \
    && dnf clean all
```

To return an error of:

```
-:5 DL3041 warning: Specify version with `dnf install -y <package>-<version>`.
```

### Actual behavior

Hadolint runs successfully with no errors.

### Steps to reproduce the behavior

Use this dockerfile:

```docker
FROM rockylinux:9.3
RUN dnf install -y dnf-plugins-core \
    && dnf clean all
```

And run hadolint against it.

This oneliner works:

```
printf "%s\n" "FROM rockylinux:9.3" "RUN dnf install -y dnf-plugins-core && dnf clean all" | docker run --rm -i hadolint/hadolint
```

If you add, e.g., `openssl` to the install command then it works as expected:

```
$ printf "%s\n" "FROM rockylinux:9.3" "RUN dnf install -y openssl dnf-plugins-core && dnf clean all" | docker run --rm -i hadolint/hadolint
-:2 DL3041 warning: Specify version with `dnf install -y <package>-<version>`.
```

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version` or
  `docker run --rm ghcr.io/hadolint/hadolint hadolint --version`:**

```bash
Haskell Dockerfile Linter 2.12.1
```

**Dockerfile** (if relevant)

```Dockerfile
FROM rockylinux:9.3
RUN dnf install -y dnf-plugins-core \
    && dnf clean all
```

**Additional environment details (OS, stack version, etc.)**

**Comment by sparrowt** [comment]
It seems this false negative applies to any package name which contains a hyphen `-`
